Mathlib/RingTheory/LaurentSeries.lean

```diff
@@ -33,10 +35,39 @@ abbrev LaurentSeries (R : Type*) [Zero R] :=
   HahnSeries ℤ R
 #align laurent_series LaurentSeries
 
-variable {R : Type u}
+variable {R : Type*}
 
 namespace LaurentSeries
 
+section Zero
+
+variable [Zero R]
+
+-- generalize to any locally finite linear order? see mathlib3 PR #18604
+theorem supp_bdd_below_supp_Pwo (f : ℤ → R) (n : ℤ) (hn : ∀(m : ℤ), m < n → f m = 0) :
+    (Function.support f).IsPwo := by
+  rw [← Set.isWf_iff_isPwo, Set.isWf_iff_no_descending_seq]
+  rintro g hg hsupp
+  refine Set.infinite_range_of_injective (StrictAnti.injective hg) ?_
+  refine (Set.finite_Icc n <| g 0).subset <| Set.range_subset_iff.2 ?_
+  simp_all only [Function.mem_support, ne_eq, Set.mem_Icc]
+  intro k
+  constructor
+  exact Int.not_lt.mp (mt (hn (g k)) (hsupp k))
+  exact (StrictAnti.le_iff_le hg).mpr (Nat.zero_le k)
+
+/-- Construct a Laurent series from any function with support that is bounded below. -/
+def LaurentFromSuppBddBelow (f : ℤ → R) (n : ℤ) (hn : ∀(m : ℤ), m < n → f m = 0) :
+    LaurentSeries R where
+  coeff := f
+  isPwo_support' := supp_bdd_below_supp_Pwo f n hn
+
+@[simp]
+theorem coeff_LaurentFromSuppBddBelow (f : ℤ → R) (m n : ℤ) (hn : ∀(m : ℤ), m < n → f m = 0) :
+    coeff (LaurentFromSuppBddBelow f n hn) m = f m := by exact rfl
```

Comment:
Make it a `dsimp` lemma:
```suggestion
    coeff (LaurentFromSuppBddBelow f n hn) m = f m := rfl
```
BTW, does `@[simps]` on the previous definition work?

Reply:
I'm not sure how to find that out, but I will investigate.

Reply:
You can put `@[simps?]` on the definition, then it'll show generated lemmas in a message. If you're happy with these lemmas, then you can remove the question mark and this lemma.